src/lib/util/util.js

```diff
@@ -98,6 +99,182 @@ class Util {
 		}
 	}
 
+	/**
+	 * Get the type of value. A better version of the `typeof` operator, basically
+	 * @param {*} value The object or primitive whose type is to be returned
+	 * @returns {string}
+	 */
+	static getType(value) {
+		if (value == null) return String(value); // eslint-disable-line eqeqeq
+		return typeof value;
+	}
+
+	/**
+	 * Get the class (constructor) name of value
+	 * @param {*} value The object whose class name is to be returned
+	 * @returns {string}
+	 */
+	static getClass(value) {
+		return value && value.constructor && value.constructor.name ?
+			value.constructor.name :
+			{}.toString.call(value).match(/\[object (\w+)\]/)[1];
+	}
+
+	/**
+	 * Get the type info for value
+	 * @param {*} value The object or primitive whose complex type is to be returned
+	 * @returns {{basicType: string, type: string}}
+	 */
+	static getComplexType(value) {
+		const basicType = this.getType(value);
+		if (basicType === 'object' || basicType === 'function') return { basicType, type: this.getClass(value) };
+		return { basicType, type: basicType };
+	}
+
+	/**
+	 * Get the arity of fn
+	 * @param {Function} fn The function whose arity is to be returned
+	 * @returns {string}
+	 */
+	static getFunctionType(fn) {
+		return `(${fn.length}-arity)`;
```

Comment:
personally not a fan of `-arity` (x) should be fine as is

Reply:
So just, e.g., "Function(2)"? I feel like that might be a little confusing.

Reply:
Technically since optionals aren't counted anyway, it's misleading for a number at all. But Function, Function(), Function(2) would be much cleaner...

Reply:
Okay.